recreate `ipfs-cache` indexeddb on `aborted` event

Tschuck opened the pull request:
Onboarding has stuck sometimes, when creating new profiles with `use-identity` flag enabled. IPFS cache kills itself by locking the whole indexeddb. Force recreating of the cache to clear the blocking of indexeddb.

- [CORE-1121]

[CORE-1121]: https://evannetwork.atlassian.net/browse/CORE-1121

Commit:
update versions.md

- [CORE-1121]

Commit:
fix ipfs-cache recreate

- [CORE-1121]

Commit:
return dbcp directly when it was loaded once before;

- [CORE-1121]

Commit:
Merge branch 'develop' into feature/CORE-1121-ipfs-cache-aborted

karladler commented on src/app/ipfs-cache.ts:
```diff
@@ -27,38 +27,103 @@ let lastCache: IPFSCache;
+          }
+        }
+      } else if (ex.number === -2147024882) {
+        // Internet Explorer 8
```
😅a good one 😂

karladler commented on src/app/ipfs-cache.ts:
```diff
@@ -27,38 +27,103 @@ let lastCache: IPFSCache;
 
-        resolve(openreq.result);
-      };
+    if (ex) {
```
`if (ex.?code)` ?
I think we can remove the IE8 branch in the if

karladler commented on src/app/ipfs-cache.ts:
```diff
@@ -27,38 +27,103 @@ let lastCache: IPFSCache;
+  private async recreateDb(): Promise<void> {
+    log('[ipfs-cache] recreate ipfs db - maybe aborted?');
+    // close previous db
+    await (await this.getOpenedDb()).close();
```
shall we check beforehand if `this.db` exists, that we don't open a DB just to close it?

karladler commented on src/app/ipfs-cache.ts:
```diff
@@ -67,7 +132,7 @@ export class IPFSCache {
    * @param      {any}     data    data to store for the ipfs hash
    */
-  async add(hash: string, data: any) {
+  async add(hash: string, data: any): Promise<any> {
```
I think set() has void return type.

```suggestion
  async add(hash: string, data: any): Promise<void> {
```

karladler commented on src/app/ipfs-cache.ts:
```diff
@@ -78,13 +143,35 @@ export class IPFSCache {
+        reject(new Error(`[ipfs-cache] - ${errorType} - ${transaction.error} - ${isQuota ? '(quota exceeded)' : ''}`));
+      };
+
+      transaction.oncomplete = (): void => resolve(request?.result);
```
where is the `request` Object actually populated? Looks to me like we would always resolve to undefined ...

karladler replied:
nvm I just saw `request = actionCallback(transaction.objectStore(this.storeName));` at the end of this function.

The structure is really confusing, can we put the part from Line 173 inside the oncomplete function? Or even better first await transaction, make the code as linear as possible.

Tschuck replied:
We cannot move it into the oncomplete, because the oncomplete is the callback handle for the request.

Commit:
implement pr feedback

- [CORE-1121]

karladler commented on src/systemjs/plugins/ens.ts:
```diff
@@ -90,7 +90,7 @@ export default function(System: any) {
       ) + '.' + devDomain
 
-      // try to load from dev domain, if is 
+      // try to load from dev domain, if is
```
there seems to be a part missing. If is "what"?

Commit:
t push
Merge branch 'feature/CORE-1121-ipfs-cache-aborted' of github.com:evannetwork/ui-dapp-browser into feature/CORE-1121-ipfs-cache-aborted

Commit:
fix comment

- [CORE-1121]

Commit:
Merge branch 'develop' into feature/CORE-1121-ipfs-cache-aborted
